runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -962,45 +1205,362 @@ TR_VectorAPIExpansion::validateVectorAliasClasses(TR_BitVector * vectorAliasTabl
+      TR_ASSERT_FATAL(_classByte128Vector, "Could not create Vector class from signature");
+
+      }
+
+   if (methodElementType == TR::Int8 && bitsLength == 128 &&
+       _classByte128Mask == NULL)
+      {
+      _classByte128Mask = comp()->fej9()->getClassFromSignature("jdk/incubator/vector/Byte128Vector$Byte128Mask", 46, owningMethod, true);
+      TR_ASSERT_FATAL(_classByte128Mask, "Could not create Mask class from signature");
+
+      }
+   }
+
+
+void
+TR_VectorAPIExpansion::boxChild(TR::TreeTop *treeTop, TR::Node *node, uint32_t i, bool checkBoxing)
+   {
+   TR::Node *child = node->getChild(i);
+
+   TR::DataType elementType;
+   int32_t bitsLength;
+   vapiObjType objectType;
+
+   if (!isVectorizedNode(child, elementType, bitsLength, objectType))
+      return;
+
+   if ((objectType != Vector /*&& objectType != Mask*/) ||
+       bitsLength != 128 ||
+       elementType != TR::Int8)
+      {
+      TR_ASSERT_FATAL(checkBoxing, "Incorrect boxing type can only be encountered during check mode");
+
+      int32_t classId = _aliasTable[child->getSymbolReference()->getReferenceNumber()]._classId;
+
+      _aliasTable[classId]._classId = -1;
+
+      if (_trace)
+         traceMsg(comp(), "Invalidated class #%d due to unsupported boxing of %d child of node %p in %s\n",
+                          classId, i, node, comp()->signature());
+
+      //TR_VerboseLog::writeLine(TR_Vlog_VECTOR_API, "Invalidated class #%d due to unsupported boxing of %d child of node %p in %s\n",
+      //                    classId, i, node, comp()->signature());
+
+      return;
+      }
+
+   if (checkBoxing) return;
+
+   TR_OpaqueClassBlock *vecClass = objectType == Vector ? _classByte128Vector : _classByte128Mask;
+
+   TR_ASSERT_FATAL(vecClass, "vecClass is NULL when boxing %p\n", child);
+
+   // generate "newarray  jitNewArray"
+   TR_OpaqueClassBlock *j9arrayClass = getArrayClassFromDataType(comp(), elementType);
+
+   int32_t elementSize = OMR::DataType::getSize(elementType);
+   int32_t numLanes = bitsLength/8/elementSize;
+
+   TR::Node *lenConst = TR::Node::iconst(node, numLanes);
+   TR::Node *typeConst = TR::Node::iconst(node, comp()->fe()->getNewArrayTypeFromClass(j9arrayClass));
+   TR::SymbolReference *symRef = comp()->getSymRefTab()->findOrCreateNewArraySymbolRef(comp()->getMethodSymbol());
+   TR::Node *newArray = TR::Node::createWithSymRef(TR::newarray, 2, lenConst, typeConst, 0, symRef);
+
+   treeTop->insertBefore(TR::TreeTop::create(comp(), TR::Node::create(TR::treetop, 1, newArray)));
+
+   // Generate vector store to the payload array
+
+   TR::VectorLength vectorLength = OMR::DataType::bitsToVectorLength(bitsLength);
+   TR::DataType opCodeType = TR::DataType::createVectorType(elementType, vectorLength);
+
+   if (objectType == Mask)
+       opCodeType = TR::DataType::createMaskType(elementType, vectorLength);
```

Comment:
Do you need to call `createVectorType()` when `objectType == Mask`?
If no, how about changing these lines to:
```
TR::DataType opCodeType = (objectType == Vector) ?
  TR::DataType::createVectorType(elementType, vectorLength) :
  TR::DataType::createMaskType(elementType, vectorLength);
```

Reply:
This code was rewritten.

Reply:
Sorry, similar code was rewritten somewhere else. Done.